src/libraries/System.IO.FileSystem/tests/Directory/SymbolicLinks.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Directory_SymbolicLinks : BaseSymbolicLinks
+    {
+        [Fact]
+        public void EnumerateDirectories_LinksWithCycles_ShouldNotThrow()
+        {
+            DirectoryInfo testDirectory = CreateDirectorySymbolicLinkToItself();
+
+            // Windows differentiates between dir symlinks and file symlinks
+            int expected = PlatformDetection.IsWindows ? 1 : 0;
+            Assert.Equal(expected, Directory.EnumerateDirectories(testDirectory.FullName).Count());
+        }
+
+        [Fact]
+        public void EnumerateFiles_LinksWithCycles_ShouldNotThrow()
+        {
+            DirectoryInfo testDirectory = CreateDirectorySymbolicLinkToItself();
+
+            // Windows differentiates between dir symlinks and file symlinks
+            int expected = PlatformDetection.IsWindows ? 0 : 1;
+            Assert.Equal(expected, Directory.EnumerateFiles(testDirectory.FullName).Count());
+        }
+
+        [Fact]
+        public void EnumerateFileSystemEntries_LinksWithCycles_ShouldNotThrow()
+        {
+            DirectoryInfo testDirectory = CreateDirectorySymbolicLinkToItself();
+            Assert.Equal(1, Directory.EnumerateFileSystemEntries(testDirectory.FullName).Count());
```

Comment:
nit: xUnit analyzer would suggest using `Single`: https://stackoverflow.com/a/46654308/5852046

```suggestion
            Assert.Single(Directory.EnumerateFileSystemEntries(testDirectory.FullName));
```

Reply:
Strangely, I did not get the suggestion. But I can change it.

Reply:
The only thing I need to confirm is if calling `Single` will force the creation of the objects, like `Count()` does, so that the disk hit is performed. If it's some kind of _lazy_ count, it won't work.